engine/src/main/java/org/terasology/rendering/opengl/LwjglRenderingProcess.java

```diff
@@ -163,7 +155,12 @@
     /* VARIOUS */
     private boolean takeScreenshot;
     private int displayListQuad = -1;
+
+    // Note: this assumes that the settings in the configs might change at runtime,
+    // but the config objects will not. At some point this might change, i.e. implementing presets.
     private Config config = CoreRegistry.get(Config.class);
+    private RenderingConfig renderingConfig = config.getRendering();
```

Comment:
I understand the concern and I agree the scenario you mention would break
things.

Let me address it differently. If i remember correctly the WorldRenderer
refreshes its config objects once per frame. Perhaps it could set the
config objects for the rendering process too, using a new setConfigs()
method.